pkg/virt-controller/services/template.go

```diff
@@ -615,6 +615,28 @@ func (t *templateService) RenderLaunchManifest(vmi *v1.VirtualMachineInstance) (
 	resources.Requests = make(k8sv1.ResourceList)
 	resources.Limits = make(k8sv1.ResourceList)
 
+	// Set Default CPUs request
+	if !vmi.IsCPUDedicated() {
+		cpuAllocationRatio := t.clusterConfig.GetCPUAllocationRatio()
+		vcpus := int64(1)
+		vmiCPUs := vmi.Spec.Domain.CPU
+		if vmiCPUs != nil {
+			vcpus = hardware.GetNumberOfVCPUs(vmiCPUs)
+			if vmiCPUs.OvercommitVCPUs != nil && *vmiCPUs.OvercommitVCPUs == false {
+				cpuAllocationRatio = 1
+			}
+		}
+
+		if vcpus != 0 && cpuAllocationRatio > 0 {
+			val := float64(vcpus) / cpuAllocationRatio
+			vcpusStr := fmt.Sprintf("%g", val)
+			if val < 0 {
+				val *= 1000
+				vcpusStr = fmt.Sprintf("%gm", val)
+			}
+			resources.Requests[k8sv1.ResourceCPU] = resource.MustParse(vcpusStr)
```

Comment:
It appears this overrides someone's defined CPU requests in the event they didn't request for dedicated resources. That might be surprising. 

I noticed with the global MemoryOvercommit option that we're setting the memory request/limits in the mutation webhook on the VMI in the event that the user didn't explicitly define their own values. Maybe that makes sense for CPU as well?

Reply:
I think that if someone has explicitly defined CPU requests, line 642 should take care of that.